Gruntfile.js

```diff
@@ -1,4 +1,14 @@
 "use strict";
+var fs = require('fs');
+var path = require('path');
+var http = require('http');
+var build = (function(){
+	// travis testing
+	if (process.env && process.env.BUILD) return process.env.BUILD == 'default' ? 'all' : 'nocompat'; 
+	// local testing
+	else return process.argv[2] == null || process.argv[2] == 'all' ? 'all' : 'nocompat';
+})();
+require('./Tests/httpServer.js')(build);
```

Comment:
The idea is ok, but the server should run as a grunt task. iirc you could use something like grunt-concurrent to run the tasks concurrently.

Now, as I understand correctly, you always run this server every time the gruntfile is loaded.